src/backend/gflow.c

```diff
@@ -154,15 +150,21 @@ STATIC void rdgenkill()
         /*      The elems are in dfo order.                     */
         /*      go.defnod[]s consist of a elem pointer and a pointer */
         /*      to the enclosing block.                         */
-        go.defnod = (DefNode *) util_calloc(sizeof(DefNode),go.deftop);
-        deftopsave = go.deftop;
+        if (go.deftop > go.defmax)
+        {
+            go.defmax = go.deftop;
+            go.defnod = (DefNode *) util_realloc(go.defnod, go.defmax, sizeof(DefNode));
```

Comment:
I assume one can be quite confident there's no escaped `go.defnod` at this point, otherwise we'll start to see memory corruption errors.